presto-main/src/main/java/com/facebook/presto/sql/planner/iterative/Rule.java

```diff
@@ -46,4 +57,9 @@ default boolean isEnabled(Session session)
 
         Session getSession();
     }
+
+    interface Result
+    {
+        Result transformTo(PlanNode planNode);
```

Comment:
I find the name confusing, for me it's rather 'createFrom' or 'fillWith' or simply 'add'. See callsite:

```
        apply(node, captures, context)
               .ifPresent(transformed -> result.transformTo(transformed));
```